### PR TITLE
Spacecat 650 iperf3 test implementation

### PR DESCRIPTION
**Pull request by fraugabel**

### CODE REVIEW

File: cloud_level_testing/features/openstack_benchmark_iperf3.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    Given I connect to OpenStack
+    Given I have deployed hosts in <network_quantity> networks
+    And I have a private key at <vm_private_ssh_key_path> for <username>
+    Then I should be able to collect all Floating IPs
+    Then I should be able to collect all VM IPs and ports
+    Then I should be able to SSH into <network_quantity> VMs and perform <conn_test> test
+   
+
+    Examples:
+	    | host_name    | vm_private_ssh_key_path  | username 	| network_quantity	| conn_test |
```
Comment (tsmado) [inline cloud_level_testing/features/openstack_benchmark_iperf3.feature:15]: Is the indentation ok here?

File: cloud_level_testing/features/steps/tools.py [hunk 2]
```diff
@@ -519,3 +560,14 @@ def run_parallel(tasks: list[tuple], timeout: int = 100) -> list[str]:
...
+def target_source_calc(jh_name, redirs, logger ,network_quantity=2):
+    for jh in range(0,network_quantity):
+        vm_quantity = len(redirs[jh_name]['vms'])
+        target_ip = redirs[jh_name]['addr']
+        source_ip = redirs[jh_name]['fip']
+        pno = redirs[jh_name]['vms'][vm_quantity-1]['port']
+        logger.log_debug(f"{jh}: vm_quantity: {vm_quantity} target_ip: {target_ip} source_ip: {source_ip} pno: {pno}")        
+        if not source_ip or not target_ip or source_ip == target_ip:
+            logger.log_debug(f"IPerf3: {source_ip}<->{target_ip}: skipped")
+        return target_ip, source_ip, pno
```
Comment (tsmado) [inline cloud_level_testing/features/steps/tools.py:573]: Again indentation seems off.

File: libs/ConnectivityClient.py [hunk 3]
```diff
@@ -289,3 +293,87 @@ def check_server_readiness(self, attempts: int, timeout: int = 10) -> bool:
...
+
+        sftp = self.client.open_sftp()
+        host = self.execute_command("hostname -I")
+        directory = self.execute_command("pwd")
+        sftp.put(scriptname,os.path.join(directory,scriptname))
+        peek=self.execute_command("ls -la")        
+        self.logger.log_debug(f"peek: {peek}")
+        self.logger.log_info(f"{scriptname} transfer completed successfully to {host}: {directory}.")
+        if sftp:
+                sftp.close()
```
Comment (tsmado) [inline libs/ConnectivityClient.py:317]: Indentation

**Review by tsmado — COMMENTED** [review]
Some formatting required

File: cloud_level_testing/features/openstack_benchmark_iperf3.feature (hunk 1 above)
Comment (fraugabel) [reply cloud_level_testing/features/openstack_benchmark_iperf3.feature:15]: resolved

File: libs/ConnectivityClient.py (hunk 3 above)
Comment (fraugabel) [reply libs/ConnectivityClient.py]: resolved

**Review by tsmado — APPROVED** [review]

**Review by piobig2871 — APPROVED** [review]

**Pull request merged**